src/pudl/glue/ferc1_eia.py

```diff
@@ -203,6 +203,36 @@ def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
             .assign(plant_table=self.table_id.value)
         )
 
+    def drop_invalid_rows(self, df):
+        """Add required valid columns before running standard drop_invalid_rows.
+
+        This parent classes' method drops the whole df if all of the
+        ``require_valid_columns`` don't exist in the df. For the glue tests only, we add
+        in empty required columns because we know that the real ETL adds columns during
+        the full transform step.
+        """
+        # ensure the required columns are actually in the df
+        list_of_lists_of_required_valid_cols = [
+            param.required_valid_cols
+            for param in self.params.drop_invalid_rows
+            if param.required_valid_cols
+        ]
+        required_valid_cols = pudl.helpers.dedupe_n_flatten_list_of_lists(
+            list_of_lists_of_required_valid_cols
+        )
+        if required_valid_cols:
+            missing_required_cols = [
+                col for col in required_valid_cols if col not in df
+            ]
```

Comment:
In general what you're really doing here is a set operation that answers the question "What elements are in `required_valid_cols` but not in `df.columns`?", which you can figure out by "subtracting" all of `df.columns` from `required_valid_cols` using `set.difference()`

```py
missing_required_cols = set(required_valid_cols).difference(df.columns)
```

Using lists preserves ordering (which isn't typically meaningful here) and will also preserve duplicate list elements, which isn't appropriate.

It's probably doing the right thing here, but I think it's better to use sets for set operations.